packages/sync/src/class-functions.php

```diff
@@ -161,22 +161,90 @@ public static function get_post_type_features() {
 	 * @return string Hosting provider.
 	 */
 	public static function get_hosting_provider() {
-		if ( defined( 'GD_SYSTEM_PLUGIN_DIR' ) || class_exists( '\\WPaaS\\Plugin' ) ) {
-			return 'gd-managed-wp';
-		}
-		if ( defined( 'MM_BASE_DIR' ) ) {
-			return 'bh';
-		}
-		if ( defined( 'IS_PRESSABLE' ) ) {
-			return 'pressable';
+		$hosting_provider_detection_methods = array(
+			'get_hosting_provider_by_known_constant',
+			'get_hosting_provider_by_known_class',
+			'get_hosting_provider_by_known_function',
+		);
+
+		$functions = new Functions();
+		foreach ( $hosting_provider_detection_methods as $method ) {
+			$hosting_provider = call_user_func( array( $functions, $method ) );
+			if ( false !== $hosting_provider ) {
+				return $hosting_provider;
+			}
 		}
-		if ( function_exists( 'is_wpe' ) || function_exists( 'is_wpe_snapshot' ) ) {
-			return 'wpe';
+
+		return 'unknown';
+	}
+
+	/**
+	 * Return a hosting provider using a set of known constants.
+	 *
+	 * @return mixed A host identifier string or false.
+	 */
+	public function get_hosting_provider_by_known_constant() {
+		$hosting_provider = false;
+
+		switch ( true ) {
+			case ( Constants::is_defined( 'GD_SYSTEM_PLUGIN_DIR' ) ):
```

Comment:
I think that it would be a bit more concise to change this `switch` to either:

* An array map of constant => hosting provider, with a loop and a return if there's a match
* An `if/else`

mostly because it isn't a real syntactic `switch` because it doesn't check for a variable.

Reply:
Good call. I've changed this. Updated previously due to not using the Constants class but this is definitely more concise with the loop.